File: app/src/main/java/com.example.safehome/Door/IDoorInfo.java
package Door;

public abstract class IDoorInfo {

	/**
	 * 
	 * @param type
	 * @param startCoordinates
	 * @param stopCoordinates
	 * @param nextWall
	 * @param symbol
	 */
	public abstract void DoorInfo(string type, string startCoordinates, string stopCoordinates, string nextWall, Symbol symbol);

	public abstract void getType();

	/**
	 * 
	 * @param Type
	 */
	public abstract void setType(DType Type);

	public abstract void getStartCoordinates();

	/**
	 * 
	 * @param startCoordinates
	 */
	public abstract void setStartCoordinates(string startCoordinates);

	public abstract void getStopCoordniates();

	/**
	 * 
	 * @param stopCoordniates
	 */
	public abstract void setStopCoordniates(string stopCoordniates);

	public abstract void getNextWall();

	/**
	 * 
	 * @param nextWall
	 */
	public abstract void setNextWall(string nextWall);

	public abstract Symbol getSymbol();

	/**
	 * 
	 * @param symbol
	 */
	public abstract void setSymbol(Symbol symbol);

}

File: app/src/main/kotlin/com.example.safehome/Door/IDoorInfo.kt
package com.example.safehome.Door;

import com.example.safehome.Door.DType;



public interface IDoorInfo {

	/**
	 * 
	 * @param type
	 * @param startCoordinates
	 * @param stopCoordinates
	 * @param nextWall
	 * @param symbol
	 */
	public void DoorInfo(String type, String startCoordinates, String stopCoordinates, String nextWall, Symbol symbol);

	public void getType();

	/**
	 * 
	 * @param Type
	 */
	public void setType(DType Type);

	public void getStartCoordinates();

	/**
	 * 
	 * @param startCoordinates
	 */
	public void setStartCoordinates(String startCoordinates);

	public void getStopCoordniates();

	/**
	 * 
	 * @param stopCoordniates
	 */
	public void setStopCoordniates(String stopCoordniates);

	public void getNextWall();

	/**
	 * 
	 * @param nextWall
	 */
	public void setNextWall(String nextWall);

	public Symbol getSymbol();

	/**
	 * 
	 * @param symbol
	 */
	public void setSymbol(Symbol symbol);

}